src/test_workflow/integ_test/process_utils.py

```diff
@@ -0,0 +1,40 @@
+class ProcessUtils:
+
+    @staticmethod
+    def terminate_process(self):
+        parent = psutil.Process(self.process.pid)
+        logging.debug("Checking for child processes")
+        child_processes = parent.children(recursive=True)
+        for child in child_processes:
+            logging.debug(f"Found child process with pid {child.pid}")
+            if child.pid != self.process.pid:
+                logging.debug(f"Sending SIGKILL to {child.pid} ")
+                child.kill()
+        logging.info(f"Sending SIGTERM to PID {self.process.pid}")
+        self.process.terminate()
```

Comment:
any reason use `terminate` not just `kill`

Reply:
Two reasons for this. One is to keep the feature parity with current code. The existing logic is using `terminate`. Second is that terminate potentially allows inclusion of other methods.